solidity/contracts/middleware/libs/InterchainAccountMirrorCalldata.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+/**
+ * Format of data:
+ * [   0: ????] destination call data
+ * [ -96:  -64] destination call value
+ * [ -64:  -32] destination ICA router
+ * [ -32:     ] destination ISM
+ */
+library InterchainAccountMirrorCalldata {
+    uint32 internal constant VALUE_OFFSET = 96;
+    uint32 internal constant ICA_ROUTER_OFFSET = 64;
```

Comment:
\*\*shiver**

dont love it still, but remains your call

Reply:
lmao
do you have an alternative?

Reply:
Adding it to the immutables of the Mirror?

Reply:
is the reason you dont like this simply the encoding is trickier to interpret? having the value be committed to in an immutable seems especially silly